src/rgw/rgw_zone.cc

```diff
@@ -1363,7 +1363,11 @@ int add_zone_to_group(const DoutPrefixProvider* dpp, RGWZoneGroup& zonegroup,
 
   const bool log_data = zonegroup.zones.size() > 1;
   for (auto& [id, zone] : zonegroup.zones) {
-    zone.log_data = log_data;
+    if (zone.tier_type == "archive") {
+      zone.log_data = false;
```

Comment:
ok, but two comments:
* in existing deployments, this wouldn't take effect until a `radosgw-admin zone modify` calls this function
* ideally this would use `RGWSyncModulesManager::supports_data_export(zone.tier_type)` so we do the right thing for other sync modules like elasticsearch and cloud sync too

unfortunately, i think that sync module stuff is specific to rados; so maybe this should all be in `RGWSyncPolicyCompat::convert_old_sync_config()`, where we have access to the 'sync module service'?

Reply:
>     * in existing deployments, this wouldn't take effect until a `radosgw-admin zone modify` calls this function

Right..Thanks! Initially I tried doing this check as part of RGWSI_Zone::do_start() using  `RGWSyncModulesManager::supports_data_export()`  but that doesn't update the zone params stored at the backend. 
So for existing deployments, do you suggest we need these changes in RGWSI_Zone::do_start() as well (in case `radosgw-admin zone modify`) is not executed? Or can we update release notes to execute the cmd post upgrade?


> 
>     * ideally this would use `RGWSyncModulesManager::supports_data_export(zone.tier_type)` so we do the right thing for other sync modules like elasticsearch and cloud sync too
> 
> 
> unfortunately, i think that sync module stuff is specific to rados; so maybe this should all be in `RGWSyncPolicyCompat::convert_old_sync_config()`, where we have access to the 'sync module service'?

Ok. I observed that at times `RGWSyncModulesManager` may not be initialized by the time we want to use  supports_data_export(..).. I will recheck in this code-path and update.

Reply:
> what do you think about leaving the zone's `log_data` flag alone, and relying exclusively on `RGWBucketSyncPolicyHandler::bucket_exports_data()` to do the right thing? that would make things easier on existing users
> 

yes..I initially started with those changes and it helps to disable the logging internally. But `bucket sync status`  and `zone_params->log_data` still shows incorrect values, which did not seem right. If that is acceptable, I will redo those changes.

> these services are a mess of circular dependencies :( from [RGWServices_Def::init()](https://github.com/ceph/ceph/blob/f5c5009/src/rgw/driver/rados/rgw_service.cc#L144), it looks like the order is zone_svc -> sync_modules_svc -> bucket_sync_sobj

Thanks for confirming.

Reply:
for `bucket sync status`, i see that [bucket_sync_status()](https://github.com/ceph/ceph/blob/f5c5009/src/rgw/rgw_admin.cc#L2848) is using `RGWBucketSyncPolicyHandler::bucket_imports_data()` to tell whether the bucket has any sources. but [bucket_source_sync_status()](https://github.com/ceph/ceph/blob/f5c5009/src/rgw/rgw_admin.cc#L2512) is using `RGWZone::syncs_from(zone_name)` which only consults the `sync_from`/`sync_from_all` fields. maybe we should use `bucket_exports_data()` there instead?

Reply:
> what do you think about leaving the zone's `log_data` flag alone, and relying exclusively on `RGWBucketSyncPolicyHandler::bucket_exports_data()` to do the right thing? that would make things easier on existing users
> 

@cbodley .. With these changes I do see that in RGWDataChangesLog::add_entry(), it returns in the below condition and thus avoids adding datalog entry 
if (!filter_bucket(dpp, bucket, y)) { 
  return 0;
}

However I still see bilog entries even for new PUT op. I do not know atm where in the code bilog entries get added but I assume if the datalog entry is not added, it shouldn't add bilog entry for that bucket opas well.  Could you please confirm.

Reply:
Hi @cbodley , I looked into updating sync policy handler to consider the zones with export disabled. But even that update is not fixing the issue. The reason being (as updated earlier), bilog seem to consider and check for only log_op flag which is always enabled for all the bucket index operations (if zone->log_data is enabled ) irrespective of whether the zone is part of the sync or not. 

Also looks like the main cause of the issue of archive zone logs not being trimmed seems to be because of this change - https://github.com/ceph/ceph/commit/b02b178f29f225e851c0e4a399eab8bd80c4aefd 

So kindly suggest if we can remove that check and allow trimming for the zones not doing export as well or should we stick to disabling logging? For the latter case, FMU we may have to update all the callers to consider the zone_svc->exports_data instead of relying on zone->log_data. Please confirm.

Reply:
https://github.com/soumyakoduri/ceph/commit/f49962bcc2e4408da1ec82a4fe0024e84cacb968 contains the changes so far done to update the sync policy handler to consider the zones with export disabled.  It does fix the radosgw-admin sync status/info commands output but doesn't address the logging issue as stated above. I will submit separate tracker/PR for the same.

Reply:
@cbodley, @mattbenjamin , @smanjara regarding the original issue, since all the zones (even with export_data disabled) will have logging enabled for all the bucket index OPs, please suggest if we can allow trimming of them ( i.e, to revert https://github.com/ceph/ceph/commit/b02b178f29f225e851c0e4a399eab8bd80c4aefd)

Reply:
thanks @soumyakoduri. for the datalog, `RGWDataChangesLog::add_entry()` is doing the right thing by checking [bucket_exports_data()](https://github.com/ceph/ceph/blob/75d3702/src/rgw/driver/rados/rgw_bucket.cc#L2299-L2301). the bucket index ops should use that same logic for their `log_op` arguments

i think it would probably make sense to pull that logic out of the datalog itself, and move it up to RGWRados functions like [bucket_index_link_olh()](https://github.com/ceph/ceph/blob/75d3702/src/rgw/driver/rados/rgw_rados.cc#L7065-L7083). there we'd have a `bool log_op = bucket_exports_data(...);` that we'd pass to the index op, and use to decide whether to call `add_datalog_entry()`

Reply:
Thanks @cbodley .. so IIU, we should update all the callers of "log_index_operation(..)" to have `bool log_op = bucket_exports_data(...);` check right?

Reply:
right. `log_index_operation()` in cls/rgw/cls_rgw.cc runs on the osd side, and `bool log_op` is a parameter for the bucket index operations in cls/rgw/cls_rgw_client.h; so we'd use `bucket_exports_data()` on the rgw side to determine what to pass for `log_op`

Reply:
Got it. Thanks!

Reply:
i was hoping that we could add the per-bucket control as well here (using `bucket_exports_data()` instead of `need_to_log_data()`), but this looks correct 👍 

@smanjara should we follow up on the per-bucket part later?

Reply:
@cbodley sure we could follow up with another pr.